pkg/proxy/iptables/proxier.go

```diff
@@ -1099,7 +1089,7 @@ func (proxier *Proxier) syncProxyRules() {
 			// If the "external" IP happens to be an IP that is local to this
 			// machine, hold the local port open so no other process can open it
 			// (because the socket might open but it would never work).
-			if (svcInfo.Protocol() != v1.ProtocolSCTP) && localAddrSet.Has(net.ParseIP(externalIP)) {
+			if (svcInfo.Protocol() != v1.ProtocolSCTP) && utilproxy.GetLocalAddrSet().Has(net.ParseIP(externalIP)) {
```

Comment:
This is 2 loop-levels down.  I don't think we want to be refreshing this every time - once per loop is sufficient.  Just read it and save it up above and then use it here?

Reply:
it is only used for ExternalIPs :shrug:

Reply:
@thockin  thanks for point out.  definng it outside the loop will be more reasonable.  
updated.